src/MerchantCenter/PriceBenchmarks.php

```diff
@@ -43,18 +234,36 @@ public function update_price_benchmarks(): void {
 			$query->reload_data();
 
 			// Insert new benchmark data.
-			foreach ( $benchmarks as $benchmark ) {
+			foreach ( $benchmarks as $id => $benchmark ) {
+				$performance_data = $this->get_performance_data( $id );
+				if ( ! empty( $performance_data ) ) {
+					// Combine metrics data into the response for the specific product.
+					$benchmark = array_merge( $benchmark, $performance_data );
+				}
+
 				$price_compared_with_benchmark = $this->price_compared_with_benchmark( $benchmark['price_micros'], $benchmark['benchmark_price_micros'] );
+				$effectiveness                 = $this->get_effectiveness( $benchmark['effectiveness'] );
 				$query->insert(
```

Comment:
This whole insert could be simplified a lot if we match and normalize the column keys during the mapping process, that way we could simply do something like: 

```php
foreach( $benchmarks as $benchmark_item ) {
  $query->insert( $benchmark_item );
}
```

Reply:
Nice catch. it look more simplified.